pkg/io/jsonio/unmarshal.go

```diff
@@ -58,21 +66,46 @@ func (*Reader) ReadSpec(filename string, configDefs *ir.ConfigDefs) (*ir.Spec, e
 
 func validateSegments(jsonssegments SpecSegments) error {
 	for _, v := range jsonssegments {
-		if v.Type != TypeSubnet {
-			return fmt.Errorf("only subnet segments are supported, not %q", v.Type)
+		if v.Type != TypeSubnet && v.Type != TypeCidr {
+			return fmt.Errorf("only subnet and cidr segments are supported, not %q", v.Type)
 		}
 	}
 	return nil
 }
 
-func translateSegments(jsonssegments SpecSegments) map[string][]string {
+func translateSegments(jsonssegments SpecSegments, segnmentType Type) map[string][]string {
 	result := make(map[string][]string)
 	for k, v := range jsonssegments {
-		result[k] = v.Items
+		if v.Type == segnmentType {
+			result[k] = v.Items
+		}
 	}
 	return result
 }
 
+func validateCidrSegments(configDefs *ir.ConfigDefs, m map[string][]string) (map[string]map[string][]string, error) {
```

Comment:
The name is unclear. Validation does not alter the data structure, so something else is going on.